core/src/main/java/org/infinispan/marshall/core/EncoderRegistryImpl.java

```diff
@@ -48,6 +55,28 @@ public void registerWrapper(Wrapper wrapper) {
    }
 
    @Override
+   public void registerTranscoder(Transcoder transcoder) {
+      transcoders.add(transcoder);
+   }
+
+   @Override
+   public Transcoder getTranscoder(MediaType mediaType, MediaType another) {
+      Optional<Transcoder> transcoder = transcoders
+            .stream()
+            .filter(t -> t.supportsConversion(mediaType, another))
+            .findAny();
+      if (!transcoder.isPresent()) {
+         throw new EncodingException("Cannot find transcoder from " + mediaType + " to " + another);
```

Comment:
I believe some of these exception deserve to be given the jboss logging treatement

Reply:
right, I will externalize them all 👍